### ISSUE TITLE
Segmentation fault: 11

### ISSUE THREAD

**Issue by dinkyrdj** [breedlun/clearplot#2]
Just installed clearplot, upgrading from BRplot, using pip. pip reported that the installation was successful except for the following:

```
/private/var/folders/yj/ny2_xfk17qz_05xxnyd5ny7r00202g/T/pip-build-HccKFn/clearplot/setup.py:57: UserWarning: WARNING: An issue occured while installing the custom fonts for clearplot.
      warnings.warn("WARNING: An issue occured while installing the custom fonts for clearplot.")
```

I tried running the Basic example from the documentation and it did not work. Here is what happens:

```
>>> import clearplot.plot_functions as pf
>>> import numpy as np
>>>
>>> x = np.arange(0,10,0.01)
>>> y = np.sqrt(x)
>>> pf.plot('bare_bones.png', x, y);
Segmentation fault: 11
```
After the segmentation fault, python exits and no file is written. Any ideas?

**Comment by breedlun** [comment]
Wow, a seg fault!  I've been using this code for years and never gotten a seg fault.  (I think the issue installing your fonts, should have nothing to do with the seg fault.)  

Can I ask for some more info about your setup?  I am guessing you are on Linux, but can you tell me what version of python, matplotlib, and numpy you are running?  Have you built your python environment from scratch, or are you using Canopy or Anaconda?

Also, do you have PySide or PyQt installed?  They are required for the Qt4Agg backend in matplotlib.  I was about to add those to the list of required packages.

**Comment by dinkyrdj** [comment]
Doh, forgot the important details, sorry about that. I'm running on Mac OS X 10.10. I'm using the Enthought Canopy python distribution. Python 2.7.9, matplotlib 1.5, and numpy 1.9.2. 

I have PyQt installed and could get PySide. Not sure if it matters, but I'm using the MacOSX backend in matplotlib.

**Comment by breedlun** [comment]
This appears to be an issue with matplotlib's Qt4Agg backend.  I have recreated the issue on my mac:

    Enthought Canopy Python 2.7.6 | 64-bit | (default, Sep 15 2014, 17:43:19) 
    [GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] on darwin
    Type "help", "copyright", "credits" or "license" for more information.
    >>> import matplotlib as mpl
    >>> mpl.rcParams['backend'] = 'Qt4Agg'
    >>> import matplotlib.pyplot as plt
    >>> import numpy as np
    >>> x = np.arange(0, 10, 0.01)
    >>> y = np.sqrt(x)
    >>> plt.plot(x,y)
    Segmentation fault: 11

The issue disappears if I use the MacOSX backend.  It also does not happen if I use Spyder instead of just using python from the command line.  I will keep looking into this.

**Comment by breedlun** [comment]
Not surprisingly, this bug in matplotlib has already been [discovered and addressed](https://github.com/matplotlib/matplotlib/issues/5687), but the new version of matplotlib has not been released yet.  (I never caught the bug in my own testing because I always use Spyder, which seems to work fine.)

The question is, what to do while we wait for matplotlib 1.5.1.  I would like to switch to the 'TkAgg' backend to remove the dependence on PySide / PyQt, but Spyder has [trouble with TkAgg](https://github.com/spyder-ide/spyder/issues/1651).  Thus, clearplot is stuck between two different bugs in open source projects.

I will see what happens if I leave the backend unspecified.

**Comment by breedlun** [comment]
Leaving the backend unspecified just defaults to Qt4Agg on my machine, so that does not solve the problem.  TkAgg is the only interactive backend that always ships with matplotlib (see [here](http://matplotlib.org/faq/usage_faq.html#what-is-a-backend) and [here](http://matplotlib.org/users/installing.html), so it makes sense to use it.  

I think I have a workaround, however, for the Spyder bug.  I can still specify the 'TkAgg' backend for general clearplot usage, and then specify the Qt4Agg backend in Spyder per [these instructions](http://clearplot.readthedocs.org/en/latest/installation.html) if I also enable the ``scientific_startup.py`` script.

**Comment by breedlun** [comment]
The seg fault issue should be resolved with the 1.0.5 release.  Please open a new issue if you wish to debug the problem with installing the custom fonts.
